fortimanager_connector.py

```diff
@@ -0,0 +1,346 @@
+        else:
+            error_text = "Error Code: {}. Error Message: {}".format(error_code, error_message)
+
+        return error_text
+
+    def _handle_test_connectivity(self, param):
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        self.save_progress("Connecting to endpoint")
+        fmg_instance = None
+
+        try:
+            fmg_instance = self._login(action_result)
+            self.save_progress("Obtaining system status")
+            response_code, response_data = fmg_instance.get('sys/status')
+            fmg_instance.logout()
+        except Exception as e:
+            error_msg = self._get_error_msg_from_exception(e)
+            self.save_progress("Test Connectivity Failed.")
+            return action_result.set_status(phantom.APP_ERROR, error_msg)
+
+        if response_code == 0:
+            self.save_progress("Test Connectivity Passed")
+            return action_result.set_status(phantom.APP_SUCCESS)
+        else:
+            self.save_progress("response_data: {}".format(response_data))
+            self.save_progress("Test Connectivity Failed.")
+            return action_result.set_status(phantom.APP_ERROR, "Test Connectivity Failed")
+
+    def _handle_create_firewall_policy(self, param):
+
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        fmg_instance = None
+        level = param['level']
+        pkg = param['package']
+        data = {
+                    'name': param['name'],
+                    'srcintf': param['source_interface'],
+                    'dstintf': param['destination_interface'],
+                    'service': param['service'],
+                    'srcaddr': param['source_address'],
+                    'dstaddr': param['destination_address'],
+                    'action': param['action'],
+                    'status': param['status'],
+                    'inspection-mode': param['inspection_mode'],
+                    'logtraffic': param['log_traffic'],
+                    'schedule': param['schedule']
+                }
+        if level == 'ADOM':
+            adom = param.get('adom')
+            if not adom:
+                adom = 'root'
+            endpoint = ADOM_FIREWALL_ENDPOINT.format(adom=adom, pkg=pkg)
+        # Global Feature TODO
+        # elif level == 'Global':
+        #     adom = 'global'
+        #     policy_type = param['policy_type']
+        #     endpoint = GLOBAL_FIREWALL_ENDPOINT.format(pkg=pkg, policy_type=policy_type)
+
+        try:
+            fmg_instance = self._login(action_result)
+            fmg_instance.lock_adom(adom)
+            response_code, response_data = fmg_instance.add(endpoint, **data)
+            fmg_instance.commit_changes(adom)
+            fmg_instance.unlock_adom(adom)
+            fmg_instance.logout()
```

Comment:
should also do these function calls outside of the `try` block here too in case `add()` or `commit_changes()` throws and exception/fails

Reply:
good eye!